WordPress/Classes/System/ReaderPresenter.swift

```diff
@@ -0,0 +1,240 @@
+import Foundation
+import UIKit
+import SwiftUI
+import Combine
+import WordPressUI
+
+/// Manages top-level Reader navigation.
+final class ReaderPresenter: NSObject, SplitViewDisplayable {
+    private let sidebarViewModel = ReaderSidebarViewModel()
+
+    // The view controllers used during split view presentation.
+    let sidebar: ReaderSidebarViewController
+    let supplementary: UINavigationController
+    var secondary: UINavigationController
+
+    /// The navigation controller for the main content when shown using tabs.
+    private let mainNavigationController = UINavigationController()
+    private var latestContentVC: UIViewController?
+
+    private var viewContext: NSManagedObjectContext {
+        ContextManager.shared.mainContext
+    }
+
+    private var cancellables: [AnyCancellable] = []
+
+    override init() {
+        secondary = UINavigationController()
+        sidebar = ReaderSidebarViewController(viewModel: sidebarViewModel)
+        sidebar.navigationItem.largeTitleDisplayMode = .automatic
+        supplementary = UINavigationController(rootViewController: sidebar)
+        supplementary.navigationBar.prefersLargeTitles = true
+
+        super.init()
+
+        sidebarViewModel.navigate = { [weak self] in self?.navigate(to: $0) }
+    }
+
+    // TODO: (reader) update to allow seamless transitions between split view and tabs
+    @objc func prepareForTabBarPresentation() -> UINavigationController {
+        sidebarViewModel.isCompact = true
+        mainNavigationController.navigationBar.prefersLargeTitles = true
+        mainNavigationController.viewControllers = [sidebar]
+        sidebar.navigationItem.backButtonDisplayMode = .minimal
+        showInitialSelection()
+        return mainNavigationController
+    }
+
+    // MARK: - Navigation
+
+    func showInitialSelection() {
+        cancellables = []
```

Comment:
If only the `selection` observer needs to be re-created, maybe we can store it separately from `cancellables`?

My concern is that other observers may get canceled by accident because other places may call `.store(in: &cancellables)`

Reply:
That's a good point. I added `selectionObserver`.